web/packages/teleport/src/User/UserContext.tsx

```diff
@@ -52,20 +63,52 @@ export function useUser(): UserContextValue {
   return useContext(UserContext);
 }
 
+// ClusterPreferences represents a map of clusterId to their cluster preferences
+type ClusterPreferences = Record<string, UserClusterPreferences>;
+
 export function UserContextProvider(props: PropsWithChildren<unknown>) {
   const { attempt, run } = useAttempt('processing');
+  const clusterPreferences = useRef<ClusterPreferences>({});
```

Comment:
can you add a comment on why we use a ref here?

Reply:
+1, I would also specify the type inline, `ClusterPreferences` doesn't give us too much.
```suggestion
  const clusterPreferences = useRef<Record<string, UserClusterPreferences>>({});
```